src/ConstructiveSolidGeometry/VolumePrimitives/Torus.jl

```diff
@@ -1,89 +1,130 @@
-struct Torus{T,TR,TB,TP,TT} <: AbstractVolumePrimitive{T}
+struct Torus{T,TR,TB,TP,TT,TZ} <: AbstractVolumePrimitive{T}
     r_torus::TR
     r_tube::TB
     φ::TP
     θ::TT
+    z::TZ
     function Torus( ::Type{T},
                    r_torus::T,
                    r_tube::Union{T, <:AbstractInterval{T}},
                    φ::Union{Nothing, <:AbstractInterval{T}},
-                   θ::Union{Nothing, <:AbstractInterval{T}}) where {T}
-        new{T,T,typeof(r_tube),typeof(φ),typeof(θ)}(r_torus, r_tube, φ, θ)
+                   θ::Union{Nothing, <:AbstractInterval{T}},
+                   z::T) where {T}
+        new{T,T,typeof(r_tube),typeof(φ),typeof(θ),T}(r_torus, r_tube, φ, θ, z)
     end
 end
 
 #Constructors
-function Torus(;r_torus = 1, r_tubeMin = 0, r_tubeMax = 1, φMin = 0, φMax = 2π, θMin = 0, θMax = 2π)
-    T = float(promote_type(typeof.((r_torus, r_tubeMin, r_tubeMax, φMin, φMax, θMin, θMax))...))
+function Torus(;r_torus = 1, r_tubeMin = 0, r_tubeMax = 1, φMin = 0, φMax = 2π, θMin = 0, θMax = 2π, z = 0)
+    T = float(promote_type(typeof.((r_torus, r_tubeMin, r_tubeMax, φMin, φMax, θMin, θMax, z))...))
     r_tube = r_tubeMin == 0 ? T(r_tubeMax) : T(r_tubeMin)..T(r_tubeMax)
     φ = mod(T(φMax) - T(φMin), T(2π)) == 0 ? nothing : T(φMin)..T(φMax)
     θ = mod(T(θMax) - T(θMin), T(2π)) == 0 ? nothing : T(θMin)..T(θMax)
-    Torus( T, T(r_torus), r_tube, φ, θ)
+    Torus( T, T(r_torus), r_tube, φ, θ, T(z))
+end
+
+Torus(r_torus, r_tubeMin, r_tubeMax, φMin, φMax, θMin, θMax, z) = Torus(;r_torus = r_torus, r_tubeMin = r_tubeMin, r_tubeMax = r_tubeMax, φMin = φMin, φMax = φMax, θMin = θMin, θMax = θMax, z = z)
+
+function Torus(r_torus::R1, r_tube::R2, z::TZ) where {R1<:Real, R2<:Real, TZ<:Real}
+    T = float(promote_type(R1, R2, TZ))
+    Torus( T, T(r_torus), T(r_tube), nothing, nothing, T(z))
+end
+
+function RoundChamfer(r_torus::R1, r_tube::R2, z::TZ) where {R1<:Real, R2<:Real, TZ<:Real}
+    T = float(promote_type(R1, R2, TZ))
+    Torus( T, T(r_torus), T(r_tube), nothing, T(0)..T(π/2), T(z))
 end
```

Comment:
The function `RoundChamfer` is never called nor is it exported. What is this?

Reply:
It's the mechanical term for a rounded corner. I was using it as a shortcut. Should we delete?

Reply:
Let us leave it. But if it is a user-convenient function, we should export it.